### PR TITLE
create stg_portal_user

### PR DESCRIPTION
**Pull request by agpmercado**
trial for PR to create stg_portal_user

### CODE REVIEW

File: models/staging/portal/stg_portal_user.sql [hunk 1]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
+        seen_delivery_modal,
+        old_pricing_rounding,
+        seen_planv2_modal,
+        feb_2021_show_sensi,
+        ignore_auto_pausing,
+        seen_oct_2021_kangaroo_modal,
+        emails_migrate_payment_reminder,
+        seen_feb_2021_update_modal,
+--excluded fields
+        _fivetran_deleted,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:58]: We should include this as a WHERE statement, i.e. where _fivetran_deleted is false

File: models/staging/portal/stg_portal_user.sql [hunk 2]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
+        _fivetran_synced
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:59]: This is good metadata, would be good to include. I would consider creating a separate category for this. Is this the equivelent of a last_updated timestamp

File: models/staging/portal/stg_portal_user.sql [hunk 3]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
---mapped fields
---excluded fields
-        id,
-        pricing_model_id,
-        shipping_address_2,
-        stripe_key,
-        shipping_state,
-        shipping_postcode,
-        old_pricing_rounding,
+        date_created,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:18]: We need agree and align on a consistent timezone designation for all timestamps. Even if timestamp is in the correct format we  should have a clear identifier or process here

File: models/staging/portal/stg_portal_user.sql [hunk 4]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
         last_login,
-        seen_planv2_modal,
-        user_type_id,
-        feb_2021_show_sensi,
+        netsuite_last_synced,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:20]: Move this to a separate metadata category

File: models/staging/portal/stg_portal_user.sql [hunk 5]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
-        seen_oct_2021_kangaroo_modal,
+        email,
+        mobile_num,
+        began_on_trial,
         trial_days,
-        seen_delivery_modal,
-        shipping_suburb_postcode_id,
         lyka_subscription,
-        stripe_invoice_prefix,
         user_type_class,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:29]: Add this to unknown category

File: models/staging/portal/stg_portal_user.sql [hunk 6]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
-        emails_migrate_payment_reminder,
-        seen_feb_2021_update_modal,
         shipping_address_1,
+        shipping_address_2,
+        shipping_state,
+        shipping_city,
+        shipping_postcode,
+        shipping_company_name,
         delivery_instructions,
+        stripe_key,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:37]: I think add this to metadata if anything. Arguably should be excluded

File: models/staging/portal/stg_portal_user.sql [hunk 7]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
+        stripe_invoice_prefix,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:38]: Same as this

File: models/staging/portal/stg_portal_user.sql [hunk 8]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
+        number_of_charges,
+        ignore_failed_payments,
+        custom_plan,
+        bab_version,
+        seen_delivery_modal,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:43]: Exclude

File: models/staging/portal/stg_portal_user.sql [hunk 9]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
         seen_dashboard_modal,
-        firstname,
-        shipping_city,
         feb_2021_show_sku,
-        mobile_num,
-        began_on_trial,
-        email,
         seen_refur_modal,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:46]: Exclude

File: models/staging/portal/stg_portal_user.sql [hunk 10]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
-        ignore_failed_payments,
         block_global_emails,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:47]: Exclude

File: models/staging/portal/stg_portal_user.sql [hunk 11]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
-        shipping_company_name,
-        custom_plan,
-        _fivetran_deleted,
-        _fivetran_synced,
         seen_fish_2022,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:48]: Exclude

File: models/staging/portal/stg_portal_user.sql [hunk 12]
```diff
@@ -9,54 +9,54 @@ renamed as (
...
-        bab_version,
-        netsuite_id,
-        netsuite_last_synced
-
+        seen_delivery_modal,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:49]: Exclude all remaining fields

**Review by alewinskylyka — CHANGES_REQUESTED** [review]

File: models/staging/portal/stg_portal_user.sql [hunk 13]
```diff
@@ -15,7 +15,7 @@ renamed as (
         shipping_suburb_postcode_id,
         netsuite_id,
 --timestamps
-        date_created,
+        date_created_au,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql]: What is the logic here with renaming this as date_created_au?

File: models/staging/portal/stg_portal_user.sql [hunk 14]
```diff
@@ -15,16 +15,13 @@ renamed as (
         shipping_suburb_postcode_id,
         netsuite_id,
 --timestamps
-        date_created_au,
-        last_login,
-        netsuite_last_synced,
+        date_created as date_created_au,
```
Comment (alewinskylyka) [inline models/staging/portal/stg_portal_user.sql:18]: Not sure what the logic is here?

**Review by alewinskylyka — APPROVED** [review]
I think this looks ok to submit to Oscar. I'm not sure about have a separate bucket called _included fields with less value_ but suggest we discuss with Oscar. More importantly we need to ascertain the source logic behind these fields.

File: models/staging/portal/stg_portal_user.sql (hunk 13 above)
Comment (agpmercado) [reply models/staging/portal/stg_portal_user.sql]: Based on the documentation you put together in Notion, this is in local time? As mentioned, I think we should align together what timezone we wanted all date time fields to be in as well as proper naming convention. This is just a placeholder until we agree on a standard

Comment (alewinskylyka) [reply models/staging/portal/stg_portal_user.sql]: Ok great lets submit the PR to Oscar for review and discuss in more detail with him when we meet next,

**Comment by agpmercado** [comment]
> I think this looks ok to submit to Oscar. I'm not sure about have a separate bucket called _included fields with less value_ but suggest we discuss with Oscar. More importantly we need to ascertain the source logic behind these fields.



> 

Yeah I agree there shouldn't be this section anyway as it is part of mapped fields but I just wanted to remember the fields we wanted to cross check with Tech so very much open to change my grouping comment! (Also I am not sure how to reply to your comment on above so just adding it here

**Comment by lukersmithoscar** [comment]
Hello, great start team. We can talk through this sometime during the week, but for the time being here is some food for thought https://www.notion.so/hello-oscar/Element-Naming-Standard-a316f64d010b41d486626213e0366990. It would be good to standardise our element naming.

**Pull request merged**